uaclient/yaml.py

```diff
@@ -0,0 +1,43 @@
+import sys
+from importlib import machinery as importlib_machinery
+from importlib import util as importlib_util
+
+from uaclient.exceptions import UserFacingError
+from uaclient.messages import MISSING_YAML_MODULE
+
+
+def get_imported_yaml_module():
+    # Maybe we already imported it?
+    if "yaml" in sys.modules:
+        return sys.modules["yaml"]
+
+    # Try to get the system yaml module
+    finder = importlib_machinery.PathFinder()
```

Comment:
Do we need to instantiate the `PathFinder` object ? it seems that `find_spec` is a class method

Reply:
You are right, will change accordingly